chef/cookbooks/monasca/recipes/ha.rb

```diff
@@ -0,0 +1,26 @@
+# Copyright 2017 FUJITSU LIMITED
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include_recipe "crowbar-pacemaker::haproxy"
+
+haproxy_loadbalancer "monasca" do
+  address "0.0.0.0"
+  port 1234
```

Comment:
```
haproxy_loadbalancer "monasca-api" do
  address network_settings 0.0.0.0 # not sure if/how we can specify multiple addresses for this resource but this will make it listen on both the admin and public interface
  port 8070
  servers CrowbarPacemakerHelper.haproxy_servers_for_service(node, "monasca", "monasca-server", "monasca-api")
  )
  action :nothing
end.run_action(:create)

haproxy_loadbalancer "monasca-log-api" do
  address network_settings 0.0.0.0 # not sure if/how we can specify multiple addresses for this resource but this will make it listen on both the admin and public interface
  port 5607
  servers CrowbarPacemakerHelper.haproxy_servers_for_service(node, "monasca", "monasca-server", "monasca-log-api")
  )
  action :nothing
end.run_action(:create)

haproxy_loadbalancer "monasca-logs-search" do
  address network_settings 0.0.0.0 # not sure if/how we can specify multiple addresses for this resource but this will make it listen on both the admin and public interface
  port 5601
  servers CrowbarPacemakerHelper.haproxy_servers_for_service(node, "monasca", "monasca-server", "monasca-logs-search")
  )
  action :nothing
end.run_action(:create)
```

For this to work you'll need to define the ports where the regular services listen in the HA case in defaults.rb:

```
default[:monasca][:ha][:ports][:monasca-api] = 18070
default[:monasca][:ha][:ports][:monasca-api] = 15607
default[:monasca][:ha][:ports][:monasca-api] = 15601
```

This is where `haproxy_servers_for_service` is defined:

https://github.com/crowbar/crowbar-ha/blob/master/chef/cookbooks/crowbar-pacemaker/libraries/helpers.rb#L244

That should make it clear why this works out. It _is_ a bit convoluted, I'm afraid.

Also, I just used the regular listen port for the `haproxy_loadbalancer`
resource's port attribute above for the sake of clarity. Please don't do this
in your cookbook. Use a helper the way the [Barbican 
barclamp](https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/barbican/recipes/ha.rb#L23)
does it. You'll find it defined here:

https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/barbican/libraries/helpers.rb

That helper ensures the actual services listen on the alternate ports (i.e.
18070, 15607, 15601 in our case) in the HA case, and on the regular ports 
(8070, 5607, 5601 in our case) otherwise. To that end it provides the network
settings in both the [service recipe](https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/barbican/recipes/api.rb#L37)
and the [HA recipe](https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/barbican/recipes/api.rb#L37).